blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/serializers/metrics/AbstractSerDes.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2016 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.rackspacecloud.blueflood.io.serializers.metrics;
+
+import com.google.protobuf.CodedInputStream;
+import com.google.protobuf.CodedOutputStream;
+import com.rackspacecloud.blueflood.io.Constants;
+import com.rackspacecloud.blueflood.types.*;
+
+import java.io.IOException;
+
+/**
+ * This is an abstract class that most Serialization/Deserialization class
+ * extends from. It contains some utility methods.
+ */
+public abstract class AbstractSerDes {
```

Comment:
I noticed that this class does not have any abstract methods. I hadnt seen such classes before but I guess if you dont want anyone to instantiate this class, this is how you do it. Also none of its children override anything from this class. So, this could very well be perceived as an utility class. 

What I would like though is,
1) either create an interface which contains methods serialize() and deserialize() 
2) or have abstract methods for serialize() and deserialize() here

I do see that its difficult to structure it that way because of classes like RawSerDes. So in summary I am not quire sure how to go about this but I feel like we arent leveraging polymorphism very well.